app/controllers/concerns/course/assessment/submission/koditsu_submissions_concern.rb

```diff
@@ -0,0 +1,196 @@
+# frozen_string_literal: true
+# rubocop:disable Metrics/ModuleLength
+module Course::Assessment::Submission::KoditsuSubmissionsConcern
+  extend ActiveSupport::Concern
+  include Course::Assessment::Question::KoditsuQuestionConcern
+
+  def fetch_all_submissions_from_koditsu(assessment)
+    @assessment = assessment
+    submission_service = Course::Assessment::Submission::KoditsuSubmissionService.new(@assessment)
+    status, response = submission_service.run_fetch_all_submissions
+
+    return [status, nil] if status != 200 && status != 207
+
+    @all_submissions = response
+    @questions = @assessment.questions.includes({ actable: :test_cases })
+
+    @order_test_cases_type = {
+      'public' => 0,
+      'private' => 1,
+      'evaluation' => 2
+    }
+
+    @test_cases_order = @questions.to_h do |question|
+      test_cases = question.actable.test_cases
+      [question.id, sort_for_koditsu(test_cases)]
+    end
+
+    @all_submissions.each do |submission|
+      next if submission['status'] == 'notStarted' || submission['status'] == 'error'
+
+      process_submission(submission)
+    end
```

Comment:
This logic triggers N+1 problem?

Reply:
understand.. right now I put the TODO for this one, to fix the code of `process_submission` so it's not triggering N+1 problem

right now should be fine as the upcoming PE will only have 30-ish students and there will be 3 problems, hence only 90 answers will be fetched from Koditsu.